deobs.py

```diff
@@ -278,14 +278,16 @@ def powershell_vars(text: bytes) -> Optional[bytes]:
     @staticmethod
     def powershell_carets(text: bytes) -> Optional[bytes]:
         """ Remove PowerShell carets """
-        output = text
-        for full in re.findall(rb'"(?:[^"]+[A-Za-z0-9]+\^[A-Za-z0-9]+[^"]+)+"', text):
-            output = output.replace(full, full.replace(b"^", b""))
-        for full in re.findall(rb'"(?:[^"]+[A-Za-z0-9]+`[A-Za-z0-9]+[^"]+)+"', output):
-            output = output.replace(full, full.replace(b"`", b""))
-        if output == text:
-            return None
-        return output
+        if b"^" in text or b"`" in text:
+            output = text
+            for full in regex.findall(rb'"(?:[^"]+[A-Za-z0-9]+\^[A-Za-z0-9]+[^"]+)+"', text):
```

Comment:
What is the point of doing these `for` loops twice? Can't we apply a single regex in `findall` that looks for both ^ and ` and then replace either character in a single iteration? And we can just check if there is a hit, and if so return formatted output.

Reply:
Would make sense to do it how you suggested since there wouldn't be any collisions during searches anyways if merged to one. This will be changed as suggested.